### PR TITLE
Add redux reducers and actions

### PR DESCRIPTION
**Pull request by MuhammadUsmanAslam**
- Add Redux Toolkit (`npm install react-redux @reduxjs/toolkit`).
- Create a directory that will contain all your Redux logic (`/src/redux`)
- Configure a Redux store (`/src/redux/store.js`)
- Define a `slice` of state for `books` that:
- - Includes an array of books (initial state: empty array)
- - Includes a reducer that adds a book
- - Includes a reducer that removes a book
- Define a slice of state for categories that:
- - Includes an array of categories (initial state: empty array)
- - Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  categoriesSlice: [],
+  isLoading: true,
+  status: 'Under Construction',
```
Comment (Qoosim) [inline src/redux/categories/categoriesSlice.js]: - [x] It is required that when the check-status button is clicked, action should return a string `Under construction`. At the initial state(first load), it should be empty string. Kindly fix that.

File: src/components/Home/Home.js [hunk 2]
```diff
@@ -4,7 +4,9 @@ import NewBookForm from './NewBookForm/NewBookForm';
 import BookList from './BookList/BookList';
 
 function Home() {
-  const [books, setBooks] = useState([{ title: 'Title 1', author: 'author 1', id: 0 }, { title: 'Title 2', author: 'author 2', id: 1 }, { title: 'Title 3', author: 'author 3', id: 2 }]);
+  const [books, setBooks] = useState([{ title: 'Title 1', author: 'author 1', id: 0 },
+    { title: 'Title 2', author: 'author 2', id: 1 },
+    { title: 'Title 3', author: 'author 3', id: 2 }]);
 
   return (
```
Comment (Qoosim) [inline src/components/Home/Home.js]: - [x] The button to remove book is not displaying when a book object is added. Kindly add `remove` button to effect the remove function.

### Your App Behavior
[app-behave.webm](https://user-images.githubusercontent.com/26597432/229728150-79793ea0-51ed-47c0-8aad-a359b5f61158.webm)

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @MuhammadUsmanAslam, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Git Flow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MuhammadUsmanAslam** [comment]
Hi @Qoosim!
Thank you for such a detailed review. 👍 
I checked your suggested changes I think those changes are not meant to be met in this project. Those are for the next project.
Regards!

**Review by addod19 — APPROVED** [review]
### Hi @MuhammadUsmanAslam, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] Nice job meeting all the requirements :rocket: 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**